src/components/screensavermanager.js

```diff
@@ -0,0 +1,134 @@
+define(["events", "playbackManager", "pluginManager"], function (events, playbackManager, pluginManager) {
+
+    function getMinIdleTime() {
+        // Returns the minimum amount of idle time required before the screen saver can be displayed
+        //return 3000;
+        return 180000;
+    }
+
+    var lastFunctionalEvent = 0;
+    function getFunctionalEventIdleTime() {
+        return new Date().getTime() - lastFunctionalEvent;
+    }
+
+    events.on(playbackManager, "playbackstop", function (e, stopInfo) {
+        var state = stopInfo.state;
+        if (state.NowPlayingItem && state.NowPlayingItem.MediaType == "Video") {
+            lastFunctionalEvent = new Date().getTime();
+        }
+    });
+
+    function ScreenSaverManager() {
+
+        var self = this;
+        var activeScreenSaver;
+
+        function showScreenSaver(screensaver) {
+
+            if (activeScreenSaver) {
+                throw new Error("An existing screensaver is already active.");
+            }
+
+            console.log("Showing screensaver " + screensaver.name);
+
+            screensaver.show();
+            activeScreenSaver = screensaver;
+
+            if (screensaver.hideOnClick !== false) {
+                window.addEventListener("click", hide, true);
+            }
+            if (screensaver.hideOnMouse !== false) {
+                window.addEventListener("mousemove", hide, true);
+            }
+            if (screensaver.hideOnKey !== false) {
+                window.addEventListener("keydown", hide, true);
+            }
+        }
+
+        function hide() {
+            if (activeScreenSaver) {
+                console.log("Hiding screensaver");
+                activeScreenSaver.hide();
+                activeScreenSaver = null;
+            }
+
+            window.removeEventListener("click", hide, true);
+            window.removeEventListener("mousemove", hide, true);
+            window.removeEventListener("keydown", hide, true);
+        }
+
+        self.isShowing = function () {
+            return activeScreenSaver != null;
+        };
+
+        self.show = function () {
+            var screensavers = pluginManager.ofType("screensaver");
+
+            require(["connectionManager"], function (connectionManager) {
+
+                var server = connectionManager.currentApiClient();
+
+                show(screensavers, server);
+            });
+        };
+
+        function show(screensavers, currentServer) {
+
+            if (currentServer) {
+
+                var loggedInScreenSavers = screensavers.filter(function (screensaver) {
+                    return !screensaver.supportsAnonymous;
+                });
+
+                if (loggedInScreenSavers.length) {
+                    screensavers = loggedInScreenSavers;
+                }
```

Comment:
why you disable anonymous screensavers when someone is not logged in?.. I get the idea of disabling non-anonymous when there's no one logged in, but why the other way around?

Reply:
https://github.com/jellyfin/jellyfin-web/pull/592/commits/4038c9bcaef22a5adabc1d375c8dc79f4d54da1b